src/Appwrite/Platform/Modules/Databases/Http/VectorDB/Embeddings/Text/Create.php

```diff
@@ -0,0 +1,145 @@
+                    responses: [
+                        new SDKResponse(
+                            code: SwooleResponse::STATUS_CODE_OK,
+                            model: $this->getBulkResponseModel(),
+                        )
+                    ],
+                    contentType: ContentType::JSON,
+                    parameters: [
+                        new Parameter('databaseId', optional: false),
+                        new Parameter('collectionId', optional: false),
+                        new Parameter('documents', optional: false),
+                    ]
+                )
+            ])
+            ->param('documents', [], fn (array $plan) => new ArrayList(new JSON(), $plan['databasesBatchSize'] ?? APP_LIMIT_DATABASE_BATCH), 'Array of documents data as JSON objects.', true, ['plan'])
+            ->inject('response')
+            ->inject('embeddingAgent')
+            ->inject('dbForProject')
+            ->inject('getDatabasesDB')
+            ->callback($this->action(...));
+    }
+
+    public function action(array $documents, UtopiaResponse $response, $embeddingAgent, Database $dbForProject, callable $getDatabasesDB): void
+    {
+        $isAPIKey = Auth::isAppUser(Authorization::getRoles());
+        $isPrivilegedUser = Auth::isPrivilegedUser(Authorization::getRoles());
+
+        if (!$isAPIKey && !$isPrivilegedUser) {
+            throw new Exception(Exception::GENERAL_BAD_REQUEST);
+        }
+
+        if (empty($documents)) {
+            throw new Exception(Exception::DOCUMENT_MISSING_DATA);
+        }
+
+        // Validate and process each document
+        $availableModels = [];
+        $availableModels = $embeddingAgent->getAdapter()->getModels();
+
+        $results = [];
+
+        // validating all documents first
+        foreach ($documents as $index => $item) {
+            if (!\is_array($item)) {
+                throw new Exception(Exception::DOCUMENT_INVALID_STRUCTURE, 'Invalid item at index ' . $index);
+            }
+
+            $text = $item['text'] ?? '';
+            $model = $item['embeddingModel'] ?? ($item['embeddingModel'] ?? null);
+
+            if (!\is_string($text) || $text === '') {
+                throw new Exception(Exception::GENERAL_ARGUMENT_INVALID, 'Missing or invalid "text" at index ' . $index);
+            }
+            if (!\is_string($model) || $model === '') {
+                throw new Exception(Exception::GENERAL_ARGUMENT_INVALID, 'Missing or invalid "embeddingModel" at index ' . $index);
+            }
+            if (!empty($availableModels) && !\in_array($model, $availableModels, true)) {
+                throw new Exception(Exception::GENERAL_BAD_REQUEST, 'Unknown embedding model: ' . $model);
+            }
+        }
+        foreach ($documents as $index => $item) {
+            $embeddingAgent->getAdapter()->setModel($model);
+
+            $embedResult = $embeddingAgent->embed($text);
```

Comment:
Let's make sure to try/catch this, and throw a new custom error (in errors.php and Exception.php) `EMBEDDING_GENERATION_FAILED` if any fail